### ISSUE TITLE
Refactor colleges functionality to provide generalized functionality

### ISSUE THREAD

**Issue by philayres** [consected/restructure#73]
## Requirements

"Colleges" provides a mechanism for users to pick an existing item from the table, or to add a new one if the desired one doesn't already exist. This forms the basis of a tagging mechanism.

The administrator should be able to configure an arbitrary dynamic model to be the source of selectable and creatable items. Another dynamic definition may define a field to point at this target dynamic model (at a specifc field) and use the same UI functionality as "colleges". The configuration may simply be an extension of the select_* fields, similar to how we specify "big-select". 

The dynamic model will already have granted a "create" user access control to enable the user to add items. If a user doesn't have "create" privileges, then they can only select from existing items in the list.

The admin console also includes the Colleges list. Consider if this is valuable for the new tag types, or whether to just use a configured report.

## Phase 2

After implementing the basic functionality, look at extending it to the multiple selection tag_select_* field types, which would then store an array of pre-existing or newly created items.

**Comment by philayres** [comment]
## Implementation Plan

Generalize the hardcoded "Colleges" select-and-create pattern into a configurable mechanism. Any dynamic model can serve as a tag source for `select_record_from_table_*` and `tag_select_record_from_table_*` fields. A new `creatable` option on the `edit_as` config enables typeahead-style text entry with auto-creation of new records, gated by user access controls. Existing Colleges functionality is left as-is (refactoring deferred).

### Phase A: Backend (2 steps)
1. **Add `creatable` config option** — new `edit_as.creatable.enabled` in the field options YAML schema (`extra_options_field_options_defs.yaml`)
2. **Auto-create callback** — `before_save` in `ImplementationHandler` iterates creatable fields, checks if value exists in target model, verifies user has `create` access, and creates the record if missing

### Phase B: Frontend Single-Select (2 steps)
3. **Extend `select_record_from_table` partial** — when `creatable.enabled`, render a typeahead text input with inline JSON data (from existing `list_record_data_for_select()`) instead of a `<select>` dropdown
4. **Generalize typeahead JS** — new init block in `_fpa_form_utils.js` targeting `[data-typeahead-for]` elements, reusing existing `setup_typeahead()` with inline JSON data

### Phase C: Multi-Select Tags (3 steps)
5. **Extend `tag_select_record_from_table` partial** — multi-value typeahead with tag chips
6. **Extend `before_save` for arrays** — iterate each value, create if missing
7. **Multi-value typeahead JS** — tag chip UI with typeahead suggestions

### Phase D: Testing (3 steps)
8. Model specs — auto-creation, access control, duplicate prevention
9. Single-select system specs — typeahead UI, create-on-save
10. Multi-select system specs — multi-tag creation, array storage

### Key design decisions
- **Value storage is configurable** via `value_attr` (text or ID)
- **UI is typeahead** (Bloodhound autocomplete), matching existing colleges UX
- **Creation is server-side** (`before_save` callback), not client-side AJAX
- **Data comes from existing `list_record_data_for_select()`** — no new definitions endpoint or SelectorCache needed
- **No admin console for tag types** initially — use configured reports
